Jellyfin.Server.Implementations/Item/BaseItemRepository.cs

```diff
@@ -0,0 +1,2183 @@
+
+namespace Jellyfin.Server.Implementations.Item;
+
+/// <summary>
+/// Handles all storage logic for BaseItems.
+/// </summary>
+public sealed class BaseItemRepository
+    : IItemRepository, IDisposable
+{
+    /// <summary>
+    /// This holds all the types in the running assemblies
+    /// so that we can de-serialize properly when we don't have strong types.
+    /// </summary>
+    private static readonly ConcurrentDictionary<string, Type?> _typeMap = new ConcurrentDictionary<string, Type?>();
+    private readonly IDbContextFactory<JellyfinDbContext> _dbProvider;
+    private readonly IServerApplicationHost _appHost;
+    private readonly IItemTypeLookup _itemTypeLookup;
+    private readonly IServerConfigurationManager _serverConfigurationManager;
+    private readonly ILogger<BaseItemRepository> _logger;
+    private bool _disposed;
+
+    private static readonly IReadOnlyList<ItemValueType> _getAllArtistsValueTypes = [ItemValueType.Artist, ItemValueType.AlbumArtist];
+    private static readonly IReadOnlyList<ItemValueType> _getArtistValueTypes = [ItemValueType.Artist];
+    private static readonly IReadOnlyList<ItemValueType> _getAlbumArtistValueTypes = [ItemValueType.AlbumArtist];
+    private static readonly IReadOnlyList<ItemValueType> _getStudiosValueTypes = [ItemValueType.Studios];
+    private static readonly IReadOnlyList<ItemValueType> _getGenreValueTypes = [ItemValueType.Studios];
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BaseItemRepository"/> class.
+    /// </summary>
+    /// <param name="dbProvider">The db factory.</param>
+    /// <param name="appHost">The Application host.</param>
+    /// <param name="itemTypeLookup">The static type lookup.</param>
+    /// <param name="serverConfigurationManager">The server Configuration manager.</param>
+    /// <param name="logger">System logger.</param>
+    public BaseItemRepository(
+        IDbContextFactory<JellyfinDbContext> dbProvider,
+        IServerApplicationHost appHost,
+        IItemTypeLookup itemTypeLookup,
+        IServerConfigurationManager serverConfigurationManager,
+        ILogger<BaseItemRepository> logger)
+    {
+        _dbProvider = dbProvider;
+        _appHost = appHost;
+        _itemTypeLookup = itemTypeLookup;
+        _serverConfigurationManager = serverConfigurationManager;
+        _logger = logger;
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        if (_disposed)
+        {
+            return;
+        }
+
+        _disposed = true;
+    }
+
+    /// <inheritdoc />
+    public void DeleteItem(Guid id)
+    {
+        ArgumentNullException.ThrowIfNull(id.IsEmpty() ? null : id);
+
+        using var context = _dbProvider.CreateDbContext();
+        using var transaction = context.Database.BeginTransaction();
+        context.PeopleBaseItemMap.Where(e => e.ItemId == id).ExecuteDelete();
+        context.Peoples.Where(e => e.BaseItems!.Count == 0).ExecuteDelete();
```

Comment:
this is inside the query. The operator is needed because the analyser cannot destingish between queries where the code is not actually executed and c# code where the operator might actually prevent NREX. 

I am not a huge fan of having a comment copy pasted 40 times all over, this is really always the same reason why this is done and it would just bloat the code without adding any meaningful info.

Reply:
thats a good alternative in my opinion

Reply:
Note added